### PR TITLE
Codemodel is now separated Maven project. 

### PR DESCRIPTION
**Pull request by bravehorsie**
Signed-off-by: Roman Grigoriadi <roman.grigoriadi@oracle.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,11 @@ The JAXB framework enables developers to perform the following operations:
 
 JAXB gives Java developers an efficient and standard way of mapping between XML and Java code. Java developers using JAXB are more productive because they can write less code themselves and do not have to be experts in XML. JAXB makes it easier for developers to extend their applications with XML and Web Services technologies.
 
+# Codemodel
+Codemodel is an independent library for java code generation, which is released separately and integrated in JAXB. 
+Besides JAXB it is used in [istack-commons-project](https://github.com/bravehorsie/jaxb-istack-commons),
```
Comment (lukasj) [inline README.md]: you don't want to point to your fork here

Comment (bravehorsie) [reply README.md]: :) ty

File: jaxb-ri/codemodel/codemodel/pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <parent>
         <groupId>com.sun.xml.bind.mvn</groupId>
         <artifactId>jaxb-codemodel-parent</artifactId>
-        <version>2.4.0-SNAPSHOT</version>
+        <version>2.3.2-SNAPSHOT</version>
```
Comment (lukasj) [inline jaxb-ri/codemodel/codemodel/pom.xml]: version should be changed in a branch, not in master

File: jaxb-ri/codemodel/codemodel/pom.xml [hunk 3]
```diff
@@ -26,46 +26,22 @@
...
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
+            <version>4.12</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>com.google.code.javaparser</groupId>
             <artifactId>javaparser</artifactId>
+            <version>1.0.11</version>
```
Comment (lukasj) [inline jaxb-ri/codemodel/codemodel/pom.xml]: this dependency is being used only here, so should be removed from former (jaxb) parent

File: jaxb-ri/codemodel/pom.xml [hunk 4]
```diff
@@ -28,5 +30,59 @@
         <module>codemodel</module>
         <module>codemodel-annotation-compiler</module>
     </modules>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.0</version>
```
Comment (lukasj) [inline jaxb-ri/codemodel/pom.xml]: I've stumbled over https://github.com/apache/maven-compiler-plugin/blob/master/src/it/multirelease-patterns/singleproject-runtime/pom.xml#L98 - maybe you could adopt it (w/o the mr related part) ?

**Comment by lukasj** [comment]
also license and notice files have to be put into binary jar, see ie https://github.com/eclipse-ee4j/common-annotations-api/blob/master/pom.xml#L164

File: jaxb-ri/codemodel/codemodel-annotation-compiler/src/main/java/com/sun/codemodel/ac/ACTask.java [hunk 5]
```diff
@@ -280,16 +269,14 @@ public void execute() throws BuildException {
...
         } finally {
-            if (userLoader != acl && userLoader instanceof Closeable) {
+            if (userLoader instanceof Closeable) {
                 try {
                     ((Closeable) userLoader).close();
                 } catch (IOException ioe) {
                     //ignore
                 }
             }
-            acl.cleanup();
```
[the remark this answers is not in the document]
Comment (bravehorsie) [reply jaxb-ri/codemodel/codemodel-annotation-compiler/src/main/java/com/sun/codemodel/ac/ACTask.java]: There is no variable acl anymore in the code.

File: jaxb-ri/codemodel/codemodel/pom.xml (hunk 2 above)
Comment (bravehorsie) [reply jaxb-ri/codemodel/codemodel/pom.xml]: right

File: jaxb-ri/codemodel/pom.xml (hunk 4 above)
Comment (bravehorsie) [reply jaxb-ri/codemodel/pom.xml]: Yes it would be good to replace current pattern in all projects with it.

**Comment by bravehorsie** [comment]
> also license and notice files have to be put into binary jar, see ie https://github.com/eclipse-ee4j/common-annotations-api/blob/master/pom.xml#L164

Codemodel doesn't have own eclipse project and own notice/license files.

File: jaxb-ri/codemodel/codemodel-annotation-compiler/src/main/java/com/sun/codemodel/ac/ACTask.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (lukasj) [reply jaxb-ri/codemodel/codemodel-annotation-compiler/src/main/java/com/sun/codemodel/ac/ACTask.java]: true. What is the type of `userLoader`?

**Comment by lukasj** [comment]
> Codemodel doesn't have own eclipse project and own notice/license files.

every binary from the repository must contain copy of the license and notice files. Use those from the repo root. Or do you think codemodel has no license at all?

File: jaxb-ri/codemodel/codemodel-annotation-compiler/src/main/java/com/sun/codemodel/ac/ACTask.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (bravehorsie) [reply jaxb-ri/codemodel/codemodel-annotation-compiler/src/main/java/com/sun/codemodel/ac/ACTask.java]: cleanup() is called from close() which is called on userLoader.

**Pull request merged**